rfcs/0005-pablo-distribution-assets/0005-pablo-distribution.adoc

```diff
@@ -0,0 +1,632 @@
+. The system MUST accumulate the rewards share for PICA holders who stake PICA token, out of the PICA supply allocated for them.
+. The system MUST support accumulating any token rewards other than PICA for PICA stakers.
+. The system must support rewards being distributed on granular basis - e.g every 6 or 12 hours.
+. The users MUST be able to claim the rewards once distributed.
+. The system SHOULD support rewards in the form of fNFTs.
+
+=== Pablo Governance
+. Governance MUST be able to set the PBLO token reward allocation.
+. Governance MUST be able to set the Pablo LP reward proportion for each Pablo LP token(i.e Pool) out of PBLO or other token reward allocation. This is to incentivize providing liquidity to required pools as decided by governance.
+. Governance MUST be able to adjust the PBLO reward rate(eg: daily) based on the incentivization strategy.
+. Pablo pool protocol fees(for rewarding protocol stakers) SHOULD be configurable as a percentage of the pool owner fee.
+
+=== PICA Governance
+. Governance MUST be able to set the PICA token reward allocation.
+. Governance MUST be able to adjust the PICA reward rate based on the incentivization strategy.
+
+=== Technical Requirements
+. The system MUST allow accumulation and mapping of rewards shares of multiple assets types(Eg: PBLO, KSM) to staked position(fNFT) type defined by another asset type(eg: PICA).
+. The system MUST support transfer of rewards using staking-rewards pallet to necessary fNFT types.
+. The system SHOULD support converting a reward accumulated in one asset type to another based on a preferred reward asset type configuration. Eg: Given a reward accumulated is in Acala it should be able to convert that to one of PBLO or PICA using the Pablo DEX pools.
+** This is to handle cases where a Pablo pool fees are in a different asset type than what is preferred.
+
+== Method
+
+=== System Overview
+
+[plantuml,images/pablo-distribution-verview,png]
+----
+skinparam component {
+  backgroundColor<<exists>> LightGreen
+  backgroundColor<<remove>> Red
+}
+
+node "Pallet-Staking-Rewards\n(Pallet-Rewards?)" {
+    StakingReward - [Reward Logic]
+    Staking - [fNFT Storage/Logic]
+    Node "BatchProcess" {
+        [RewardDisbursementHook]
+        [RewardsConfig] --> [RewardAccumulationHook]
+    }
+}
+
+[fNFT Storage/Logic] <<exists>>
+[RewardDisbursementHook] <<remove>>
+
+node "Pallet-Pablo" {
+    [Pool] --> StakingReward : transfer\ntrading fees\nfor PBLO stakers
+    [Pool] --> Staking : stake LP tokens
+    [Pool] --> [Pool] : accumulate LP\ntrading fees
+    [FeeConfig] --> [Pool]
+}
+
+[Pool] <<exists>>
+
+
+node "Governance" {
+    [Configuration] --> [FeeConfig]
+    [Treasury]--> StakingReward: Transfer LP, PBLO\nor other token rewards
+}
+----
+
+TODO: What to do for part of protocol fees that should be transferred to treasury eventually as treasury does not stake it's PBLO?
```

Comment:
can remove it. as RFC does not decides on it. 

may be it should not stake. afaik treasury is used in emergencies of some kinds or rewarding. 

so kind of remove TODO, to RFC would be final.